cupyx/cutensor.pyx

```diff
@@ -423,9 +431,9 @@ cdef inline ContractionDescriptor _create_contraction_descriptor(
     cdef ContractionDescriptor desc
 
     key = (handle.ptr, cutensor_compute_type,
-           desc_A.ptr, mode_A.data, alignment_req_A,
-           desc_B.ptr, mode_B.data, alignment_req_B,
-           desc_C.ptr, mode_C.data, alignment_req_C)
+           desc_A.ptr, mode_A, alignment_req_A,
+           desc_B.ptr, mode_B, alignment_req_B,
+           desc_C.ptr, mode_C, alignment_req_C)
```

Comment:
I feel it could be error prone to cache pointers that CuPy does not own (just like the modes), so perhaps we shouldn't use `desc.ptr` here either? Let me look into that.

Reply:
For tensor descriptors, `create_tensor_descriptor()` would cache the created descriptor, so `desc.ptr` would not be invalidated. Perhaps we should do the same thing in `create_modes()` and cache the `Mode` object, then `mode.data` would stay valid too.

Reply:
Done 8af1d6f.